mfl_test.go

```
@@ -222,6 +222,45 @@ func TestJSONParseSliceAndMap(t *testing.T) {
 	}
 }
 
+// parse() must decode \uXXXX JSON escapes (UTF-8 encoding the code point) --
+// #311: the old parser dropped the backslash and leaked the literal "uXXXX"
+// text through, silently corrupting any JSON with an escaped non-ASCII char
+// (common: LLM API responses escaping an em-dash, curly quote, etc).
+func TestJSONParseUnicodeEscape(t *testing.T) {
+	got := runProg(t,
+		`type T struct { s string }`,
+		`func main() { t := parse("{\"s\":\"a\\u000bb\\u2014d\"}", T{}) println(t.s) }`)
+	want := "ab—d\n"
```

golangci-lint (2.12.2)

[error] ST1018: string literal contains the Unicode control character U+000B, consider using the '\v' escape sequence instead

(staticcheck)